src/Dropdown/Dropdown.js

```diff
@@ -395,23 +396,26 @@ class Dropdown extends Util.mixin(BindMixin) {
       );
     }
 
-    if (state.renderHidden) {
-      dropdownMenu = (
-        <div key="concealer" className="dropdown-menu-concealer">
-          {dropdownMenu}
-        </div>
-      );
-    }
+    // if (state.renderHidden) {
+    //   dropdownMenu = (
+    //     <div key="concealer" className="dropdown-menu-concealer">
+    //       {dropdownMenu}
+    //     </div>
+    //   );
+    // }
 
     if (props.transition) {
       dropdownMenu = (
-        <CSSTransitionGroup
-          transitionName={transitionName}
-          transitionEnterTimeout={props.transitionEnterTimeout}
-          transitionLeaveTimeout={props.transitionLeaveTimeout}
+        <CSSTransition
+          in={state.isOpen && !state.renderHidden}
```

Comment:
That was the important bit. I guess we can move `!state.renderHidden` to the line 407 otherwise it transitions while being hidden